.github/copilot-instructions.md

```diff
@@ -0,0 +1,238 @@
+# IDE Sidecar - AI Coding Agent Instructions
+
+## Project Overview
+
+IDE Sidecar is a **Java 21 Quarkus application** that serves as the backend for Confluent's VS Code extension. It exposes REST and GraphQL APIs for managing connections to Kafka clusters and consuming/producing messages. The project is compiled to a **native executable using GraalVM** for fast startup and low memory footprint.
+
+**Key Connection Types:**
+- `CCLOUD`: Confluent Cloud connections (OAuth-based authentication)
+- `LOCAL`: Confluent Local/standalone Kafka
+- `DIRECT`: Direct connections to Kafka clusters (supports multiple auth methods: Basic, SASL/SCRAM, Kerberos, OAuth, TLS)
+
+## Architecture
+
+### Processor Chain Pattern
+
+Request handling uses a **chain-of-responsibility pattern** via the `Processor<T, U>` class. Processors are chained using `Processor.chain(...)`:
+
+```java
+// Example from MessageViewerProcessorBeanProducers
+Processor.chain(
+    new ConnectionProcessor<>(connectionManager),
+    new KafkaClusterInfoProcessor<>(clusterCache),
+    new ConsumeStrategyProcessor(nativeStrategy, ccloudStrategy),
+    new EmptyProcessor<>()
+)
+```
+
+Each processor:
+1. Receives a context object (e.g., `KafkaRestProxyContext`, `ProxyContext`)
+2. Performs its operation (auth, cluster lookup, proxying)
+3. Passes the modified context to the next processor
+
+**Find processor implementations in:** `src/main/java/io/confluent/idesidecar/restapi/processors/` and `src/main/java/io/confluent/idesidecar/restapi/proxy/`
```

Comment:
`src/main/java/io/confluent/idesidecar/restapi/processors/` holds only the abstract class `Processor`. Implementations can be found in `src/main/java/io/confluent/idesidecar/restapi/proxy/`.